File: frontend/ProjectManagement.Core/Models/CreateWorkItemRequest.cs
namespace ProjectManagement.Core.Models;

public sealed record CreateWorkItemRequest
{
    public required Guid ProjectId { get; init; }
    public required WorkItemType ItemType { get; init; }
    public required string Title { get; init; }
    public string? Description { get; init; }
    public Guid? ParentId { get; init; }
    public string Status { get; init; } = "backlog";
    public string Priority { get; init; } = "medium";
    public Guid? AssigneeId { get; init; }
    public int? StoryPoints { get; init; }
    public Guid? SprintId { get; init; }
}

File: backend/pm-server/src/api/work_items/create_work_item_request.rs
use serde::Deserialize;

/// Request body for creating a work item
#[derive(Debug, Deserialize)]
pub struct CreateWorkItemRequest {
    pub project_id: String,
    pub item_type: String,
    pub title: String,
    #[serde(default)]
    pub description: Option<String>,
    #[serde(default)]
    pub parent_id: Option<String>,
    #[serde(default)]
    pub status: Option<String>,
    #[serde(default)]
    pub priority: Option<String>,
}
